src/check_nesting.cpp

```diff
@@ -0,0 +1,53 @@
+#include "sass.hpp"
+#include <vector>
+
+#include "check_nesting.hpp"
+#include "context.hpp"
+
+namespace Sass {
+
+  CheckNesting::CheckNesting(Context& ctx)
+  : ctx(ctx),
+    block_stack(std::vector<Block*>())
+  { }
+
+  Statement* CheckNesting::operator()(Block* b)
+  {
+    Block* bb = SASS_MEMORY_NEW(ctx.mem, Block, b->pstate(), b->length(), b->is_root());
```

Comment:
IMHO if it should never mutate than the `append_block` is also superflous. Doesn't it always replace the children with itself anyway under that condition? Otherwise good work!